components/net/filemanager_thread.rs

```diff
@@ -85,6 +92,19 @@ impl FileManager {
         }).expect("Thread spawning failed");
     }
 
+    pub fn fetch_file(&self,
+                     sender: servo_channel::Sender<Data>,
+                     id: Uuid,
+                     check_url_validity: bool,
+                     origin: FileOrigin,
+                     response: &mut Response) {
+        let store = self.store.clone();
+        let mut r2 = response.clone();
+        thread::Builder::new().name("read file".to_owned()).spawn(move || {
+            store.try_fetch_file(&sender, id, check_url_validity, origin, &mut r2)
```

Comment:
If try_fetch_file returns an Err, we shouldn't ignore it. I propose we check for an Err and set the response's body to Done like https://github.com/servo/servo/blob/master/components/net/http_loader.rs#L1160-L1163 instead.